woocommerce/woocommerce
# Update AdvancedFilters to use createInterpolateElement instead of interpolateComponents

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR updates the `AdvancedFilters` component in `@woocommerce/components` to use `createInterpolateElement` from `@wordpress/element` instead of `interpolateComponents` from `@automattic/interpolate-components`.

Usage of `AdvancedFilters` in `plugins/woocommerce-admin` has also been updated. Specially, the following screens are affected:

- `WooCommerce` > `Customers`
- `Analytics` > `Orders`
- `Analytics` > `Variations`
- `Analytics` > `Downloads`

Each of those screens has a `Show` > `Advanced filters` option that uses `AdvancedFilters`.

**No functionality has been changed. The screens above should behavior exactly as they did before.**

For example, here is what the `Analytics` > `Orders` advanced filters look like:

<img width="1276" alt="Screenshot 2023-04-25 at 15 01 38" src="https://user-images.githubusercontent.com/2098816/234377137-60e76633-01bb-4049-adec-c14a2c014b17.png">

Part of #37312.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Verify Storybook stories look correct:
    - `pnpm --filter=@woocommerce/storybook storybook`
    - There are some slight styling issues not due to this PR... as long as filter controls appear and testing of actual product screens looks correct, these can be ignored
    - `AdvancedFilters` > `Basic`
    - `ReportFilters` > `Examples ` > `Advanced filters`
2. Verify the use of advanced filters on `WooCommerce` > `Customers`
3. Verify the use of advanced filters on `Analytics` > `Orders`
4. Verify the use of advanced filters on `Analytics` > `Variations`
5. Verify the use of advanced filters on `Analytics` > `Downloads`
